src/pokenav.c

```diff
@@ -51,142 +51,158 @@ static void Task_RunLoopedTask(u8 taskId);
-        .unk10 = sub_81C99C0,
-        .unk14 = sub_81C9430,
-        .unk18 = sub_81C99D4,
+        .init = PokenavCallback_Init_ConditionSearchMenu,
+        .callback = GetMenuHandlerCallback,
+        .open = OpenPokenavMenuNotInitial,
+        .createLoopTask = CreateMenuHandlerLoopedTask,
+        .isLoopTaskActive = IsMenuHandlerLoopedTaskActive,
+        .free1 = FreeMenuHandlerSubstruct1,
+        .free2 = FreeMenuHandlerSubstruct2,
     },
+    [POKENAV_MAIN_MENU_CURSOR_ON_MATCH_CALL - POKENAV_MENU_IDS_START] = 
     {
-        .unk0 = PokenavCallback_Init_4,
-        .unk4 = sub_81C941C,
-        .unk8 = sub_81C9940,
-        .unkC = CreateMenuHandlerLoopedTask,
-        .unk10 = sub_81C99C0,
-        .unk14 = sub_81C9430,
-        .unk18 = sub_81C99D4,
+        .init = PokenavCallback_Init_MainMenuCursorOnMatchCall,
+        .callback = GetMenuHandlerCallback,
+        .open = OpenPokenavMenuNotInitial,
+        .createLoopTask = CreateMenuHandlerLoopedTask,
+        .isLoopTaskActive = IsMenuHandlerLoopedTaskActive,
+        .free1 = FreeMenuHandlerSubstruct1,
+        .free2 = FreeMenuHandlerSubstruct2,
     },
+    [POKENAV_MAIN_MENU_CURSOR_ON_RIBBONS - POKENAV_MENU_IDS_START] = 
     {
-        .unk0 = PokenavCallback_Init_5,
-        .unk4 = sub_81C941C,
-        .unk8 = sub_81C9940,
-        .unkC = CreateMenuHandlerLoopedTask,
-        .unk10 = sub_81C99C0,
-        .unk14 = sub_81C9430,
-        .unk18 = sub_81C99D4,
+        .init = PokenavCallback_Init_MainMenuCursorOnRibbons,
+        .callback = GetMenuHandlerCallback,
+        .open = OpenPokenavMenuNotInitial,
+        .createLoopTask = CreateMenuHandlerLoopedTask,
+        .isLoopTaskActive = IsMenuHandlerLoopedTaskActive,
+        .free1 = FreeMenuHandlerSubstruct1,
+        .free2 = FreeMenuHandlerSubstruct2,
     },
+    [POKENAV_REGION_MAP - POKENAV_MENU_IDS_START] = 
     {
-        .unk0 = PokenavCallback_Init_6,
-        .unk4 = sub_81CC554,
-        .unk8 = sub_81CC5F4,
-        .unkC = sub_81CC62C,
-        .unk10 = sub_81CC65C,
-        .unk14 = sub_81CC524,
-        .unk18 = sub_81CC670,
+        .init = PokenavCallback_Init_RegionMap,
+        .callback = GetRegionMapCallback,
+        .open = OpenPokenavRegionMap,
+        .createLoopTask = CreateRegionMapLoopedTask,
+        .isLoopTaskActive = IsRegionMapLoopedTaskActive,
+        .free1 = FreeRegionMapSubstruct1,
+        .free2 = FreeRegionMapSubstruct2,
     },
+    [POKENAV_CONDITION_PARTY - POKENAV_MENU_IDS_START] = 
     {
-        .unk0 = PokenavCallback_Init_7,
-        .unk4 = sub_81CD070,
-        .unk8 = sub_81CDDD4,
-        .unkC = sub_81CDE2C,
-        .unk10 = sub_81CDE64,
-        .unk14 = sub_81CD1C0,
-        .unk18 = sub_81CECA0,
+        .init = PokenavCallback_Init_7,
+        .callback = sub_81CD070,
+        .open = sub_81CDDD4,
+        .createLoopTask = sub_81CDE2C,
+        .isLoopTaskActive = sub_81CDE64,
+        .free1 = sub_81CD1C0,
+        .free2 = sub_81CECA0,
```

Comment:
Name these (and below entries)

Reply:
They're not related to region map, and I haven't clarified the distinction between the Conditions/Ribbons menus yet (e.g. `POKENAV_MENU_9` or `POKENAV_MENU_A`). I'll do the rest of these when I continue with the rest of pokenav.